src/ui/views/Table/TableView.svelte

```diff
@@ -121,85 +197,141 @@
-      onColumnConfigure={(column, editable) => {
-        const field = fields.find((field) => field.name === column.field);
-
-        if (field) {
-          new ConfigureFieldModal(
+          return null;
+        }}
+        onRowAdd={() => {
+          new CreateNoteModal($app, project, (name, templatePath, project) => {
+            api.addRecord(createDataRecord(name, project), templatePath);
+          }).open();
+        }}
+        onRowEdit={(id, values) => {
+          new EditNoteModal(
             $app,
-            $i18n.t("modals.field.configure.title"),
-            field,
-            editable,
-            (field) => {
-              if (editable) {
-                if (field.name !== column.field) {
-                  api.updateField(field, column.field);
-                } else {
-                  api.updateField(field);
+            fields,
+            (record) => {
+              api.updateRecord(record, fields);
+            },
+            {
+              id,
+              values,
+            }
+          ).open();
+        }}
+        onRowDelete={(id) => api.deleteRecord(id)}
+        onColumnHide={(column) => handleVisibilityChange(column.field, false)}
+        onColumnConfigure={(column, editable) => {
+          const field = fields.find((field) => field.name === column.field);
+
+          if (field) {
+            new ConfigureFieldModal(
+              $app,
+              $i18n.t("modals.field.configure.title"),
+              field,
+              editable,
+              (field) => {
+                if (editable) {
+                  if (field.name !== column.field) {
+                    api.updateField(field, column.field);
+                  } else {
+                    api.updateField(field);
+                  }
                 }
-              }
 
-              const projectFields = Object.fromEntries(
-                Object.entries(project.fieldConfig ?? {}).filter(([key, _]) =>
-                  fields.find((field) => field.name === key)
-                )
-              );
-
-              if (field.typeConfig) {
-                settings.updateProject({
-                  ...project,
-                  fieldConfig: {
-                    ...projectFields,
-                    [field.name]: field.typeConfig,
-                  },
-                });
+                const projectFields = Object.fromEntries(
+                  Object.entries(project.fieldConfig ?? {}).filter(([key, _]) =>
+                    fields.find((field) => field.name === key)
+                  )
+                );
+
+                if (field.typeConfig) {
+                  settings.updateProject({
+                    ...project,
+                    fieldConfig: {
+                      ...projectFields,
+                      [field.name]: field.typeConfig,
+                    },
+                  });
+                  saveConfig({ ...config });
```

Comment:
Project level field configs, like rich-text or options, and view-level configs like column order and column width are stored separatedly. When updating project config, the view config would be overwritten.

`project` and `config` are both local variables. Updating project config by local `project` would ignore changed `config`. Updating view config by local `config` right after updating project config here would keep the view config unchanged.